substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -2246,23 +2252,42 @@ pub mod pallet {
 			let member_account = T::Lookup::lookup(member_account)?;
 			let mut member =
 				PoolMembers::<T>::get(&member_account).ok_or(Error::<T>::PoolMemberNotFound)?;
-			let current_era = T::Staking::current_era();
+			let current_era = T::StakeAdapter::current_era();
 
 			let bonded_pool = BondedPool::<T>::get(member.pool_id)
 				.defensive_ok_or::<Error<T>>(DefensiveError::PoolNotFound.into())?;
 			let mut sub_pools =
 				SubPoolsStorage::<T>::get(member.pool_id).ok_or(Error::<T>::SubPoolsNotFound)?;
 
+			let slash_weight =
```

Comment:
quesiton for you to think about: how as this not an issue in the past? 

probable answer: because slashes were immediate, and now they are more lazy?

Reply:
The slashes were system initiated before. It still is for the pool, but yeah we did not had lazy slashes before.